webpack/react_app/components/TargetingHosts/TargetingHosts.js

```diff
@@ -0,0 +1,92 @@
+import React, { useState, useEffect } from 'react';
+import API from 'foremanReact/API';
+import PropTypes from 'prop-types';
+import HostItem from './HostItem/HostItem.js';
+
+const TargetingHosts = (props) => {
+  const { jobInvocationId } = props;
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const [hosts, setHosts] = useState([]);
+
+  const renderLoading = () => {
+    if (loading) {
+      return (
+        <tr>
+          <td colSpan="3" className="text-center">
+            Loading hosts ...
+          </td>
+        </tr>
+      );
+    }
+
+    return (null);
+  };
+
+  const getHostsData = () => {
+    const apiUrl = `/job_invocations/${jobInvocationId}?format=json`;
+
+    API.get(apiUrl)
```

Comment:
can you move this logic into a redux action?

Reply:
Is it necessary? Honestly don't have much experience (or zero) with Redux, and I don't see any bonus in connecting this small component to Redux. Can we keep it like that for now?

Reply:
I think we already using redux in this project, you can even reuse the API middleware from core, and reduce more boilerplate.
If you want we can work on it together offline